### PR TITLE
Refactor/dynamite/codestyle

### PR DESCRIPTION
**Pull request by Leptopoda**

### CODE REVIEW

**Comment by Leptopoda** [comment]
It's still far from perfect but I think it's already a lot cleaner.

If you don't like the helper methods on the models we can use extension methods for them.

**Comment by Leptopoda** [comment]
It's gonna be hard to review I know; sorry in advance.
I didn't have any plans and I just found more motivation to clean it up more and more.

The only behavior change should be unifying the method order and style of `anyOf`s and the normal objects.

---

The generation time also went down from 45s to 35s (just the build_runner without any further fixing and formatting)

File: packages/dynamite/dynamite/lib/src/models/operation.dart [hunk 1]
```diff
@@ -41,4 +44,14 @@ class Operation {
   final Responses? responses;
 
   final List<SecurityRequirement>? security;
+
+  Iterable<String> get formatedDescription sync* {
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/models/operation.dart:48]:
```suggestion
  Iterable<String> get formattedDescription sync* {
```

File: packages/dynamite/dynamite/lib/src/models/schema.dart [hunk 2]
```diff
@@ -83,17 +86,21 @@ class Schema {
   final bool? nullable;
 
   bool get isContentString => type == 'string' && (contentMediaType?.isNotEmpty ?? false) && contentSchema != null;
+
+  Iterable<String> get formatedDescription => descriptionToDocs(description);
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/models/schema.dart:90]:
```suggestion
  Iterable<String> get formattedDescription => descriptionToDocs(description);
```

File: packages/dynamite/dynamite/lib/src/models/tag.dart [hunk 3]
```diff
@@ -15,4 +18,6 @@ class Tag {
   final String name;
 
   final String? description;
+
+  Iterable<String> get formatedDescription => descriptionToDocs(description);
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/models/tag.dart:22]:
```suggestion
  Iterable<String> get formattedDescription => descriptionToDocs(description);
```

File: packages/dynamite/dynamite/lib/src/helpers/typeresult.dart
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/helpers/typeresult.dart:1]: File should be named `type_result.dart`

File: packages/dynamite/dynamite/lib/src/builder/client.dart [hunk 4]
```diff
@@ -0,0 +1,651 @@
...
+                  const Code('body = await response.body;'),
+                  const Code('} on FormatException {'),
+                  const Code("body = 'binary';"),
+                  const Code('}'),
+                  const Code(''),
+                  Code('return ${state.classPrefix}ApiException('),
+                  const Code('response.statusCode,'),
+                  const Code('response.responseHeaders,'),
+                  const Code('body,'),
+                  const Code(');'),
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/builder/client.dart:94]: Can you change this to a single object? This is really not readable :sweat_smile:

File: packages/dynamite/dynamite/lib/src/builder/client.dart [hunk 5]
```diff
@@ -0,0 +1,653 @@
...
+                ..initializers.add(
+                  const Code('''
+                          super(
+                            client.baseURL,
+                            baseHeaders: client.baseHeaders,
+                            httpClient: client.httpClient,
+                            cookieJar: client.cookieJar,
+                            authentications: client.authentications,
+                          )
+                        '''),
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/builder/client.dart:200]: Indentation is off

File: packages/dynamite/dynamite/lib/src/builder/client.dart [hunk 6]
```diff
@@ -0,0 +1,653 @@
...
+                  .expand((final element) => element)
+                  .toSet()
+                  .join(',') ??
+              '';
+          final code = StringBuffer('''
+                            var _path = '${pathEntry.key}';
+                            final _queryParameters = <String, dynamic>{};
+                            final _headers = <String, String>{${acceptHeader.isNotEmpty ? "'Accept': '$acceptHeader'," : ''}};
+                            Uint8List? _body;
+                          ''');
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/builder/client.dart:312]: Indentation is off

File: packages/dynamite/dynamite/lib/src/builder/client.dart [hunk 7]
```diff
@@ -0,0 +1,653 @@
...
+          final securityRequirements = security.where((final requirement) => requirement.isNotEmpty);
+          final isOptionalSecurity = securityRequirements.length != security.length;
+          code.write('    // coverage:ignore-start\n');
+          for (final requirement in securityRequirements) {
+            final securityScheme = spec.components!.securitySchemes![requirement.keys.single]!;
+            code.write('''
+                                if (${isRootClient ? 'this' : '_rootClient'}.authentications.where((final a) => a.type == '${securityScheme.type}' && a.scheme == '${securityScheme.scheme}').isNotEmpty) {
+                                  _headers.addAll(${isRootClient ? 'this' : '_rootClient'}.authentications.singleWhere((final a) => a.type == '${securityScheme.type}' && a.scheme == '${securityScheme.scheme}').headers);
+                                }
+                              ''');
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/builder/client.dart:324]: Indentation is off

File: packages/dynamite/dynamite/lib/src/builder/client.dart [hunk 8]
```diff
@@ -0,0 +1,653 @@
...
+            if (securityRequirements.last != requirement) {
+              code.write('else');
+            }
+          }
+          if (securityRequirements.isNotEmpty && !isOptionalSecurity) {
+            code.write('''
+                                else {
+                                  throw Exception('Missing authentication for ${securityRequirements.map((final r) => r.keys.single).join(' or ')}');
+                                }
+                              ''');
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/builder/client.dart:334]: Indentation is off

File: packages/dynamite/dynamite/lib/src/builder/client.dart [hunk 9]
```diff
@@ -0,0 +1,653 @@
...
+              nullable: dartParameterNullable,
+            ).dartType;
+
+            if (result.name == 'String') {
+              if (parameter.schema?.pattern != null) {
+                code.write('''
+                                  if (!RegExp(r'${parameter.schema!.pattern!}').hasMatch(${toDartName(parameter.name)})) {
+                                    throw Exception('Invalid value "\$${toDartName(parameter.name)}" for parameter "${toDartName(parameter.name)}" with pattern "\${r'${parameter.schema!.pattern!}'}"'); // coverage:ignore-line
+                                  }
+                                  ''');
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/builder/client.dart:361]: Indentation is off

File: packages/dynamite/dynamite/lib/src/builder/client.dart [hunk 10]
```diff
@@ -0,0 +1,653 @@
...
+              }
+              if (parameter.schema?.minLength != null) {
+                code.write('''
+                                  if (${toDartName(parameter.name)}.length < ${parameter.schema!.minLength!}) {
+                                    throw Exception('Parameter "${toDartName(parameter.name)}" has to be at least ${parameter.schema!.minLength!} characters long'); // coverage:ignore-line
+                                  }
+                                  ''');
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/builder/client.dart:368]: Indentation is off

File: packages/dynamite/dynamite/lib/src/builder/client.dart [hunk 11]
```diff
@@ -0,0 +1,653 @@
...
+              }
+              if (parameter.schema?.maxLength != null) {
+                code.write('''
+                                  if (${toDartName(parameter.name)}.length > ${parameter.schema!.maxLength!}) {
+                                    throw Exception('Parameter "${toDartName(parameter.name)}" has to be at most ${parameter.schema!.maxLength!} characters long'); // coverage:ignore-line
+                                  }
+                                  ''');
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/builder/client.dart:375]: Indentation is off

File: packages/dynamite/dynamite/lib/src/builder/client.dart [hunk 12]
```diff
@@ -0,0 +1,653 @@
...
+          code.write(
+            '''
+                            final _response = await ${isRootClient ? 'this' : '_rootClient'}.doRequest(
+                              '$httpMethod',
+                              Uri(path: _path, queryParameters: _queryParameters.isNotEmpty ? _queryParameters : null).toString(),
+                              _headers,
+                              _body,
+                            );
+                          ''',
+          );
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/builder/client.dart:498]: Indentation is off

File: packages/dynamite/dynamite/lib/src/builder/header_serializer.dart [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+                const Code('final key = iterator.current! as String;'),
+                const Code('iterator.moveNext();'),
+                const Code('final value = iterator.current! as String;'),
+                const Code('switch (key) {'),
+                ...deserializeProperty(state, identifier, spec, schema),
+                const Code('}'),
+                const Code('}'),
+                const Code(''),
+                const Code('return result.build();'),
+              ]);
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/builder/header_serializer.dart:102]: This could also be converted into a single object to make it more readable.

**Review by provokateurin — CHANGES_REQUESTED** [review]
Super awesome cleanup! :heart:

I think we should settle on a style/indentation for the inline code. We could align the whole code to the left and remove the extra indentation from the real code around it. I'm not sure if that will look good, but it would at least be unified and easier to handle if the code indentation changes later on.

**Comment by Leptopoda** [comment]
is this how you imagined it?

**Comment by provokateurin** [comment]
Yes, perfect :)

**Review by provokateurin — APPROVED** [review]

**Pull request merged**